numpy/linalg/_linalg.py

```diff
@@ -3306,6 +3306,7 @@ def vector_norm(x, /, *, axis=None, keepdims=False, ord=2):
 
     """
     x = asanyarray(x)
+    shape = list(x.shape)
```

Comment:
Maybe just define `ndim = x.ndim` since you don't need the shape otherwise.

Or, maybe better but slightly more involved, do `shape = x.shape` (no need for the list) and replace `x.shape` below with `shape` -- though that can be simplified further to `.reshape((-1, *[shape[i] for i in rest]))`

Reply:
We do need `shape = list(x.shape)` for the code below. We could probably rewrite it to not require that, but it would be a more intrusive change.

Reply:
Duh, sorry, I completely missed that you just moved the definition of `shape` up.